src/ansible/roles/client/tasks/main.yml

```diff
@@ -82,3 +82,22 @@
     else
       exit 0
     fi
+
+- name: Copy random.c to /opt
+  copy:
+    src: '{{ playbook_dir }}/../tools/random.c'
+    dest: /opt/random.c
+    owner: root
+    group: root
+    mode: 0644
+  when: (ansible_facts['distribution'] == "Fedora") or
```

Comment:
Use new fact as well.

Reply:
The file should be removed after compilation.

You can also wrap it with `block` together with compilation.

Reply:
Done both of the improvements